File: template.m
#import <Foundation/Foundation.h>

BOOL isDebugEnabled = YES;
BOOL isReadFromFile = YES;
NSString *dataFile = @"template.txt";

void debug(NSString *formatString, ...)
{
  if (!isDebugEnabled) {
    return;
  }

  NSString *contents;
  va_list args;
  va_start(args, formatString);
  contents = [[[NSString alloc] initWithFormat:formatString arguments:args] autorelease];
  va_end(args);

  NSLog(@"%@", contents);
}

void print(NSString *formatString, ...) {
  NSString *contents;
  va_list args;
  va_start(args, formatString);
  contents = [[[NSString alloc] initWithFormat:
    [formatString stringByAppendingString:@"\n"] arguments:args] autorelease];
  va_end(args);

  [[NSFileHandle fileHandleWithStandardOutput] writeData:
    [contents dataUsingEncoding: NSUTF8StringEncoding]];
}

NSArray *readLinesFromStdin()
{
  NSData *data = [[NSFileHandle fileHandleWithStandardInput] readDataToEndOfFile];
  NSString *contents = [[[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding] autorelease];
  NSArray *tmpLines = [contents componentsSeparatedByString:@"\n"];

  NSMutableArray *lines = [@[] mutableCopy];
  for (NSString *row in tmpLines) {
    if ([row length] > 0) {
      [lines addObject: row];
    }
  }

  return [lines copy];
}

NSArray *readLinesFromFile()
{
  NSError *error = nil;
  NSString *contents = [NSString stringWithContentsOfFile: dataFile
    encoding:NSUTF8StringEncoding error:&error];

  if (error != nil) {
    [NSException raise:@"data file can't read." format:@"%@", error];
  }

  NSArray *tmpLines = [contents componentsSeparatedByString:@"\n"];
  NSMutableArray *lines = [@[] mutableCopy];

  for (NSString *row in tmpLines) {
    if ([row length] > 0) {
      [lines addObject: row];
    }
  }

  return [lines copy];
}

NSMutableArray * linesToListOfStrList(NSArray *lines)
{
  NSMutableArray *matrix = [[[NSMutableArray alloc] init] autorelease];

  for (NSString *line in lines) {
    NSArray *row = [line componentsSeparatedByString:@" "];
    [matrix addObject: row];
  }

  return matrix;
}

NSMutableArray * linesToListOfIntList(NSArray *lines)
{
  NSMutableArray *matrix = [[[NSMutableArray alloc] init] autorelease];

  for (NSString *line in lines) {
    NSArray *tmpRow = [line componentsSeparatedByString:@" "];
    NSMutableArray *row = [@[] mutableCopy];
    for (NSString *cell in tmpRow) {
      [row addObject:[NSNumber numberWithInt:[cell intValue]]];
    }
    [matrix addObject: [row copy]];
  }

  return matrix;
}

int main(int argc, char** argv)
{
  NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];

  NSArray *inputLines;
  if (isReadFromFile) {
    inputLines = readLinesFromFile();
  } else {
    inputLines = readLinesFromStdin();
  }
  debug(@"inputLines=%@", inputLines);

  NSArray *inputValues = linesToListOfIntList(inputLines);
  debug(@"inputValues=%@", inputValues);

  [pool release];

  return 0;
}

File: template.php
<?php

$is_debug_enabled = true;
$is_read_from_file = true;
$data_file = "template.txt";

function debug($msg) {
  global $is_debug_enabled;
  if ($is_debug_enabled) {
    echo $msg . "\n";
  }
}

function dump($obj) {
  global $is_debug_enabled;
  if ($is_debug_enabled) {
    var_dump($obj);
  }
}

function read_lines_from_file() {
  global $data_file;
  $data = file_get_contents($data_file);
  $lines = preg_split("/\r\n|\r|\n/", $data);

  $lines = array_filter(
    $lines,
    function($x) {
      return strlen($x) > 0;
    }
  );

  return $lines;
}

function read_lines_from_stdin() {
  global $data_file;
  $data = file_get_contents("php://stdin");
  $lines = preg_split("/\r\n|\r|\n/", $data);

  $lines = array_filter(
    $lines,
    function($x) {
      return strlen($x) > 0;
    }
  );

  return $lines;
}

function lines_to_list_of_list($lines) {
  $values = array();
  foreach($lines as $row) {
    $values[] = preg_split("/ /", $row);
  }

  return $values;
}

function lines_to_list_of_int_list($lines) {
  $values = array();
  foreach($lines as $row) {
    $values[] = array_map(function($x) {
      return intval($x);
    }, preg_split("/ /", $row));
  }

  return $values;
}

function main() {
  global $is_read_from_file;

  if ($is_read_from_file) {
    $input_lines = read_lines_from_file();
  } else {
    $input_lines = read_lines_from_stdin();
  }

  dump($input_lines);

  $input_values = lines_to_list_of_list($input_lines);
  dump($input_values);

}

main();
?>
